flake-modules/dev/launch-test.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+: "${NIXVIM_NIX_COMMAND:=nix}"
+if [[ -z $NIXVIM_SYSTEM ]]; then
+	NIXVIM_SYSTEM=$(nix eval --raw --impure --expr "builtins.currentSystem")
+fi
+
+help() {
+	cat <<EOF
+Usage: tests [OPTIONS] [tests...]
+
+If tests are passed on the command line only these will be launched
+
+Options:
+	-h/--help: Display this help message and exit
+	-l/--list: Display the list of tests and exit
+	-i/--interactive: Pick interactively the tests. Can't be supplied if tests where passed.
+EOF
+}
+
+if ! OPTS=$(getopt -o "hli" -l "help,list,interactive" -- "$@"); then
+	echo "Invalid options" >&2
+	help
+	exit 1
+fi
+
+eval set -- "$OPTS"
+
+specified_tests=()
+interactive=false
+
+mk_test_list() {
+	nix eval ".#checks.${NIXVIM_SYSTEM}" --apply builtins.attrNames --json |
+		jq -r 'map(select(startswith("test-")))[]'
+}
+
+while true; do
+	case "$1" in
+	-h | --help)
+		help
+		exit 0
+		;;
+	-l | --list)
+		mk_test_list
+		exit 0
+		;;
+	-i | --interactive)
+		interactive=true
+		shift
+		;;
+	--)
+		shift
+		specified_tests=("$@")
+		break
+		;;
```

Comment:
I think we also need to handle mixed opts+args using `*)`.

Anything handled with `*)` should be added to the `specified_tests` list, and then everything after `--)` should also be added (without overwriting).

Reply:
No need to handle that, `getopt` already does it for us!

Reply:
`getopt` normalizes the arguments in the form `[OPTIONS] -- [POSITIONAL ARGS]`